rosidl_runtime_py/set_message.py

```diff
@@ -33,9 +33,15 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
     :param values: The values to set in the ROS message. The keys of the dictionary represent
         fields of the message.
     :raises AttributeError: If the message does not have a field provided in the input dictionary.
-    :raises ValueError: If a message value does not match its field type.
+    :raises TypeError: If a message value does not match its field type.
     """
-    for field_name, field_value in values.items():
+    try:
+        items = values.items()
+    except AttributeError:
+        raise TypeError(
+            "Value '%s' is expected to be a dictionary but is a %s" %
```

Comment:
I think we more specifically call these "fields", so:

```suggestion
            "Field '%s' is expected to be a dictionary but is a %s" %
```

Reply:
Since the message actually outputs the `values` the term "Field" isn't correct.

Reply:
But what is in the '%s' *is* the field name.  The most correct is:

```
Value for field '%s' is expected to be a dictionary but is a %s.
```

It's a little wordy, but maybe that is better.

Reply:
No, it is not. When I run this patch for the above stated command I get the following error message:

> Value 'now' is expected to be a dictionary but is a str

And "not" is the value of the field "stamp".

Reply:
Ah sorry, I got confused.  You are right, it is the actual value.